### PR TITLE
Doctor

### PR DESCRIPTION
**Pull request by GantMan**
new doctor

image

### CODE REVIEW

File: packages/ignite/src/plugins/default/commands/doctor.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const R = require('ramda')
+const Shell = require('shelljs')
+
+module.exports = async function (context) {
+  try {
```
Comment (skellock) [inline packages/ignite/src/plugins/default/commands/doctor.js]: As we chatted offline, `gluegun` can provide a helper mode to not swallow exceptions that you can drive from `--dev`.  Thanks for the suggestion.  Will make that for `0.7.0`.

File: packages/ignite/src/plugins/default/commands/doctor.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    const { debug } = context.print
+
+    // ... be the CLI you wish to see in the world
+    const platform = process.platform
+    const ignitePath = R.trim(Shell.which('ignite') || '')
```
Comment (skellock) [inline packages/ignite/src/plugins/default/commands/doctor.js]: I'll get a `context.system.which` for ya!

File: packages/ignite/src/plugins/default/commands/doctor.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    const igniteVersion = R.trim(Shell.exec('ignite -V', { silent: true }).stdout)
```
Comment (skellock) [inline packages/ignite/src/plugins/default/commands/doctor.js]: Another thing you can do is:

```js
const igniteVersion = R.trim(await context.system.run('ignite -V'))
```

File: packages/ignite/src/plugins/default/commands/doctor.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    const nodePath = R.trim(Shell.which('node'))
+    const npmVersion = R.trim(Shell.exec('npm --version', { silent: true }).stdout)
+    const npmPath = R.trim(Shell.which('npm'))
+    const nodeVersion = R.trim(Shell.exec('node --version', { silent: true }).stdout)
+    const ggPackageFile = `${process.cwd()}/node_modules/gluegun/package.json`
+    const ggVersion = Shell.test('-f', ggPackageFile) ? require(ggPackageFile).version : '¯\\_(ツ)_/¯'
```
Comment (skellock) [inline packages/ignite/src/plugins/default/commands/doctor.js]: Another option:

```js
const { filesystem: { exists, read } } = context

const ggVersion = exists(ggPackageFile) ? read(ggPackageFile, 'json').version : '¯\\_(ツ)_/¯'
```

File: packages/ignite/src/plugins/default/commands/doctor.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  Path: ${npmPath}
+Gluegun
+  Version: ${ggVersion}
+React Native CLI
+  Version: ${rnCli}
+App
+  React Native Version: ${appReactNativeVersion}
+\`\`\`
+`
+    debug(body, 'Ignite Info')
```
Comment (skellock) [inline packages/ignite/src/plugins/default/commands/doctor.js]: `context.print.info` is the "just print it" style.... but rainbows are pretty awesome too!

File: packages/ignite/src/plugins/default/commands/doctor.js (hunk 1 above)
Comment (GantMan) [inline packages/ignite/src/plugins/default/commands/doctor.js]: I forgot this guy, woops

File: packages/ignite/src/plugins/default/commands/doctor.js (hunk 4 above)
Comment (GantMan) [inline packages/ignite/src/plugins/default/commands/doctor.js]: can we have `'¯\\_(ツ)_/¯'` stored in context.print with checkmark etc?

File: packages/ignite/src/plugins/default/commands/doctor.js (hunk 5 above)
Comment (GantMan) [inline packages/ignite/src/plugins/default/commands/doctor.js]: 🦄 🦄 🦄 🦄 🦄 🦄 🦄 🦄 🦄 🦄 🦄 
image
🦄 🦄 🦄 🦄 🦄 🦄 🦄 🦄 🦄 🦄 🦄

**Pull request merged**